index.js

```diff
@@ -43,32 +45,46 @@ export default class QRCodeScanner extends Component {
     reactivateTimeout: 0,
     fadeIn: true,
     showMarker: false,
+    emptyView: <View />,
   }
 
   constructor(props) {
     super(props);
     this.state = {
       scanning: false,
-      fadeInOpacity: new Animated.Value(0)
+      fadeInOpacity: new Animated.Value(0),
+      isAuthorised: false,
     }
 
     this._handleBarCodeRead = this._handleBarCodeRead.bind(this);
   }
 
+  componentWillMount() {
+    if (Platform.OS === 'ios') {
+      Camera.checkVideoAuthorizationStatus().then(isAuthorised => {
+        this.setState({ isAuthorised })
```

Comment:
nit: can we change it to `isAuthorized` with a `z` instead. Just want to have it consistent with the native apis.